bokeh/plotting/helpers.py

```diff
@@ -247,6 +247,30 @@ def _tool_from_string(name):
 
         raise ValueError("unexpected tool name '%s', %s tools are %s" % (name, text, nice_join(matches)))
 
+
+def _process_axis_and_grid(plot, axis_type, axis_location, minor_ticks, axis_label, rng, dim):
+    axiscls = _get_axis_class(axis_type, rng)
+    if axiscls:
+
+        if axiscls is LogAxis:
+            plot.x_mapper_type = 'log'
+
+        # this is so we can get a ticker off the axis, even if we discard it
+        axis = axiscls(plot=plot if axis_location else None)
+
+        if hasattr(axis.ticker, 'num_minor_ticks'):
```

Comment:
Yeah, I wasn't aware of this class my self, just `hasattr()`  used for this purpose makes me look for alternatives.